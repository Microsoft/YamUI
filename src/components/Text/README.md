# Text

The `Text` component simply renders a `<span>`. It offers size and color props so UI features don't need to own this CSS. This is both a convenience for engineers and a way to enforce consistency of supported text colors and font-size/line-height combinations.

## Properties

| Prop | Type | Description |
| ---- | ---- | ----------- |
| size | TextSize | Sets both font-size and line-height, ensuring text conforms to our vertical rhythm. |
| color | TextColor | Sets the text color to one of the supported values. |

## Notes for use

`<Text>` is an inline element. Note that inline elements cannot set a smaller line-height than what is set or inherited by the nearest block-level parent element. For this reason, it is generally best to set font-size on a parent block-level element (`<Block textSize={TextSize.LARGE}>`) rather than on individual `<Text>` coponents.

## Examples

### Basic usage

```js
<Text color={TextColor.PRIMARY} size={TextSize.LARGE}>
  This example has the primary black text color with size LARGE
</Text>
```
