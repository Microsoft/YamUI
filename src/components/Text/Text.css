/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
@import "src/css/variables/colors.css";
@import "src/css/variables/fonts.css";


.y-text {
  display: inline-block;
}

.y-text__color-primary {
  color: $textColor;
}
.y-text__color-secondary {
  color: $textColor__secondary;
}
.y-text__color-metadata {
  color: $textColor__metadata;
}
.y-text__color-white {
  color: $textColor__white;
}
