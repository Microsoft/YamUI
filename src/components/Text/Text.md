The `Text` component simply renders a `span`. It offers size and color props so UI features don't need to own this CSS. This is both a convenience for engineers and a way to enforce consistency of supported text colors and font-size/line-height combinations.

### Notes for use

`Text` is an inline element. Note that inline elements cannot set a smaller line-height than what is set or inherited by the nearest block-level parent element. For this reason, it is generally best to set a `textSize` on a parent `Block` component rather than on individual `Text` components.

### Examples

Basic text:

```js { "props": { "data-example": "basic" } }
<div>
  <p>
    <Text>
      The <strong>&lt;Text /&gt;</strong> component renders its children in a <strong>span</strong>. It's a convenient way to render a piece of text with a specific size. Text will support more options in the future.
    </Text>
  </p>
  <p>
    <Text>
      By default the <strong>&lt;Text /&gt;</strong> component does not render any styling so it can inherit styles of its parent.
    </Text>
  </p>
</div>
```

Different text sizes:

```js { "props": { "data-example": "sizes" } }
const { TextSize } = require('.');

<div>
  <p>
    <Text>
      This is text without a size specified. It will inherit its size and line-height from its parent.
    </Text>
  </p>
  <p>
    <Text size={TextSize.XXLARGE}>This is size <em>XXLARGE</em>.</Text>
  </p>
  <p>
    <Text size={TextSize.XLARGE}>This is size <em>XLARGE</em>.</Text>
  </p>
  <p>
    <Text size={TextSize.LARGE}>This is size <em>LARGE</em>.</Text>
  </p>
  <p>
    <Text size={TextSize.MEDIUM}>This is size <em>MEDIUM</em>.</Text>
  </p>
  <p>
    <Text size={TextSize.MEDIUM_SUB}>This is size <em>MEDIUM_SUB</em>.</Text>
  </p>
  <p>
    <Text size={TextSize.SMALL}>This is size <em>SMALL</em>.</Text>
  </p>
  <p>
    <Text size={TextSize.XSMALL}>This is size <em>XSMALL</em>.</Text>
  </p>
</div>
```

Different text sizes with long text:

```js { "props": { "data-example": "sizes long" } }
const { TextSize } = require('.');

<div>
  <p>
    <Text>
      This is text without size specified. It will inherit its size and line-height from its parent. This is text without size specified. <strong>It will inherit</strong> its size and line-height from its parent. This is text without size specified. It will inherit its size and line-height from its parent.
    </Text>
  </p>
  <p>
    <Text size={TextSize.XXLARGE}>
      This is size <em>XXLARGE</em>. This is size <em>XXLARGE</em>. This is size <em>XXLARGE</em>. This is size <em>XXLARGE</em>. This is size <em>XXLARGE</em>. This is size <em>XXLARGE</em>. <strong>This is size <em>XXLARGE</em>.</strong> This is size <em>XXLARGE</em>. This is size <em>XXLARGE</em>. This is size <em>XXLARGE</em>. This is size <em>XXLARGE</em>. This is size <em>XXLARGE</em>.
    </Text>
  </p>
  <p>
    <Text size={TextSize.XLARGE}>
      This is size <em>XLARGE</em>. This is size <em>XLARGE</em>. This is size <em>XLARGE</em>. This is size <em>XLARGE</em>. This is size <em>XLARGE</em>. This is size <em>XLARGE</em>. <strong>This is size <em>XLARGE</em>.</strong> This is size <em>XLARGE</em>. This is size <em>XLARGE</em>. This is size <em>XLARGE</em>. This is size <em>XLARGE</em>. This is size <em>XLARGE</em>.
    </Text>
  </p>
  <p>
    <Text size={TextSize.LARGE}>
      This is size <em>LARGE</em>. This is size <em>LARGE</em>. This is size <em>LARGE</em>. This is size <em>LARGE</em>. This is size <em>LARGE</em>. This is size <em>LARGE</em>. <strong>This is size <em>LARGE</em>.</strong> This is size <em>LARGE</em>. This is size <em>LARGE</em>. This is size <em>LARGE</em>. This is size <em>LARGE</em>. This is size <em>LARGE</em>.
    </Text>
  </p>
  <p>
    <Text size={TextSize.MEDIUM}>
      This is size <em>MEDIUM</em>. This is size <em>MEDIUM</em>. This is size <em>MEDIUM</em>. This is size <em>MEDIUM</em>. This is size <em>MEDIUM</em>. This is size <em>MEDIUM</em>. <strong>This is size <em>MEDIUM</em>.</strong> This is size <em>MEDIUM</em>. This is size <em>MEDIUM</em>. This is size <em>MEDIUM</em>. This is size <em>MEDIUM</em>. This is size <em>MEDIUM</em>.
    </Text>
  </p>
  <p>
    <Text size={TextSize.MEDIUM_SUB}>
      This is size <em>MEDIUM_SUB</em>. This is size <em>MEDIUM_SUB</em>. This is size <em>MEDIUM_SUB</em>. This is size <em>MEDIUM_SUB</em>. This is size <em>MEDIUM_SUB</em>. This is size <em>MEDIUM_SUB</em>. <strong>This is size <em>MEDIUM_SUB</em>.</strong> This is size <em>MEDIUM_SUB</em>. This is size <em>MEDIUM_SUB</em>. This is size <em>MEDIUM_SUB</em>. This is size <em>MEDIUM_SUB</em>. This is size <em>MEDIUM_SUB</em>.
    </Text>
  </p>
  <p>
    <Text size={TextSize.SMALL}>
      This is size <em>SMALL</em>. This is size <em>SMALL</em>. This is size <em>SMALL</em>. This is size <em>SMALL</em>. This is size <em>SMALL</em>. This is size <em>SMALL</em>. <strong>This is size <em>SMALL</em>.</strong> This is size <em>SMALL</em>. This is size <em>SMALL</em>. This is size <em>SMALL</em>. This is size <em>SMALL</em>. This is size <em>SMALL</em>.
    </Text>
  </p>
  <p>
    <Text size={TextSize.XSMALL}>
      This is size <em>XSMALL</em>. This is size <em>XSMALL</em>. This is size <em>XSMALL</em>. This is size <em>XSMALL</em>. This is size <em>XSMALL</em>. This is size <em>XSMALL</em>. <strong>This is size <em>XSMALL</em>.</strong> This is size <em>XSMALL</em>. This is size <em>XSMALL</em>. This is size <em>XSMALL</em>. This is size <em>XSMALL</em>. This is size <em>XSMALL</em>.
    </Text>
  </p>
</div>
```
