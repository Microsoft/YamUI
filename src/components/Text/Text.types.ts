/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { NestableBaseComponentProps } from '../../util/BaseComponent/props';

export enum TextColor {
  /**
   * Primary black text color
   */
  PRIMARY = 'primary',

  /**
   * Secondary dark gray text color
   */
  SECONDARY = 'secondary',

  /**
   * Tertiary color for metadata
   */
  METADATA = 'metadata',

  /**
   * For use on darker backgrounds
   */
  WHITE = 'white',

  /**
   * For use in error messages.
   */
  ERROR = 'error',

  /**
   * For use in interaction elements which have become disabled.
   */
  DISABLED = 'disabled',
}

export enum TextSize {
  /**
   * Large headers. Font-size 2.4rem, line-height 2.8rem.
   */
  XXLARGE = 'xxLarge',

  /**
   * Group headers. Font-size 2.2rem, line-height 2.8rem.
   */
  XLARGE = 'xLarge',

  /**
   * Promo headlines. Font-size 1.8rem, line-height 2.4rem.
   */
  LARGE = 'large',

  /**
   * Body text, default size. Font-size 1.5rem, line-height 2.0rem.
   */
  MEDIUM = 'medium',

  /**
   * Secondary text. Font-size 1.4rem, line-height 2.0rem.
   */
  MEDIUM_SUB = 'mediumSub',

  /**
   * Metadata. Font-size 1.2rem, line-height 1.6rem.
   */
  SMALL = 'small',

  /**
   * Very limited use. Font-size 1.0rem, line-height 1.6rem.
   */
  XSMALL = 'xSmall',
}

export interface TextProps extends NestableBaseComponentProps {
  /**
   * Text color
   */
  color?: TextColor;

  /**
   * A preset size which determines a font-size + line-height combination
   * supporting our vertical rhythm.
   */
  size?: TextSize;

  /**
   * Sets a max-width on the Text content, hiding the overflow with an ellipsis character.
   * You should generally use a px value, or 100%.
   */
  maxWidth?: string;

  /**
   * Sets font-weight: bold.
   */
  bold?: boolean;

  /**
   * Sets text to uppercase.
   */
  uppercase?: boolean;

  /**
   * If provided, will hide child content from screenreaders using aria-hidden while making
   * the given screenreaderText available to screenreaders.
   */
  screenreaderText?: string;
}
