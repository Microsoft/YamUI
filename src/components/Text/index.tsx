/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { Enum } from 'typescript-string-enums';
import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
import { TextSize } from '../../util/enums/textSize';
import './text.css';

export const TextColor = Enum({
  /**
   * Primary black text color
   */
  PRIMARY: 'primary',

  /**
   * Secondary dark gray text color
   */
  SECONDARY: 'secondary',

  /**
   * Tertiary color for metadata
   */
  METADATA: 'metadata',

  /**
   * For use on darker backgrounds
   */
  WHITE: 'white',
});
export type TextColor = Enum<typeof TextColor>;

export interface TextProps extends NestableBaseComponentProps {
  size?: TextSize;
  color?: TextColor;
}

export { TextSize };

// TODO: Create a helper to handle this across components
const getClasses = (props: TextProps) => {
  const classes: string[] = ['y-text'];
  if (props.size) {
    classes.push(`y-text__size-${props.size}`);
  }
  if (props.color) {
    classes.push(`y-text__color-${props.color}`);
  }
  if (props.className) {
    classes.push(props.className);
  }
  return classes.join(' ');
};

const Text: React.StatelessComponent<TextProps> = props => (
  <span className={getClasses(props)}>{props.children}</span>
);
export default Text;
