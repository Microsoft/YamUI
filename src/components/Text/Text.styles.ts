/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { TextSize, TextColor } from './Text.types';
import { memoizeFunction } from 'office-ui-fabric-react/lib/Utilities';
import { mergeStyleSets, ITheme } from 'office-ui-fabric-react/lib/Styling';
import { ellipsisStyle, fontWeights, textColors, verticalAligns } from '../../util/styles/fonts';

export interface TextClassNameProps {
  color?: TextColor;
  size?: TextSize;
  maxWidth?: string;
  bold?: boolean;
  uppercase?: boolean;
  theme: ITheme;
}

const iconSizeForSize = {
  [TextSize.XXLARGE]: '2.4rem',
  [TextSize.XLARGE]: '2.2rem',
  [TextSize.LARGE]: '1.5rem',
  [TextSize.MEDIUM]: '1.4rem',
  [TextSize.MEDIUM_SUB]: '1.4rem',
  [TextSize.SMALL]: '1.2rem',
  [TextSize.XSMALL]: '1.0rem',
};

export const getClassNames = memoizeFunction((classNameProps: TextClassNameProps) => {
  const { size, maxWidth, bold, uppercase, color, theme } = classNameProps;
  const font = size ? theme.fonts[size === TextSize.MEDIUM_SUB ? 'smallPlus' : size] : undefined;

  return mergeStyleSets({
    root: {
      ...(maxWidth ? ellipsisStyle : {}),
      fontWeight: bold ? fontWeights.bold : undefined,
      display: 'inline-block',
      textTransform: uppercase ? 'uppercase' : undefined,
      fontSize: font ? font.fontSize : undefined,
      lineHeight: font ? font.lineHeight : undefined,
      color: color ? textColors(theme)[color] : undefined,
      maxWidth: maxWidth || undefined,
      height: maxWidth && font ? font.lineHeight : undefined,
      verticalAlign: maxWidth ? '-0.4rem' : undefined,
      selectors: {
        '&.y-text .y-icon': {
          top: size === TextSize.XSMALL || size === TextSize.SMALL ? '0.1rem' : undefined,
        },
        '.y-icon': {
          height: size ? iconSizeForSize[size] : undefined,
          width: size ? iconSizeForSize[size] : undefined,
        },
        /* increased specificity to override the block style */
        '&.y-text.y-text__ellipsis': {
          verticalAlign: maxWidth && size ? verticalAligns[size] : undefined,
        },
      },
    },
  });
});
