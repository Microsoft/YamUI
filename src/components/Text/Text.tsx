/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
import { TextColor, TextSize } from './enums';
import './Text.css';

export { TextColor, TextSize };

export interface TextProps extends NestableBaseComponentProps {
  /**
   * Text color
   */
  color?: TextColor;
  /**
   * A preset size which determines a font-size + line-height combination
   * supporting our vertical rhythm.
   */
  size?: TextSize;
}

/**
 * A `Text` component simply renders a `span`. It offers size and color props so UI features don't
 * need to own this CSS. This is both a convenience for engineers and a way to enforce consistency
 * of supported text colors and `font-size`/`line-height` combinations.
 */
export default class Text extends React.PureComponent<TextProps, {}> {
  render() {
    const { children } = this.props;

    return <span className={this.getClasses()}>{children}</span>;
  }

  private getClasses() {
    const { className, color, size } = this.props;

    const classes: string[] = ['y-text'];
    if (size) {
      classes.push(`y-textSize-${size}`);
    }
    if (color) {
      classes.push(`y-text__color-${color}`);
    }
    if (className) {
      classes.push(className);
    }

    return classes.join(' ');
  }
}
