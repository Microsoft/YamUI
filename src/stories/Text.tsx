/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import Text, { TextColor, TextSize } from '../components/Text';
import withReadme from 'storybook-readme/with-readme';
const readme = require('../components/Text/README');

storiesOf('Text', module)
  .addDecorator(withReadme(readme))
  .add('without props', () => (
    <div>
      <Text>
        The <b>&lt;Text /&gt;</b> component renders its children in a span. It's a convenient way to
        render a piece of text with a specific size. Text will support more options in the future.
      </Text>
      <br />
      <Text>
        By default the <b>&lt;Text /&gt;</b> component does not render any styling so it can inherit
        styles of its parent.
      </Text>
    </div>
  ))
  .add('sizes', () => (
    <div>
      <Text>
        This is text without a size specified. It will inherit its size and line-height from its
        parent.
      </Text>
      <br />
      <Text size={TextSize.XXLARGE}>This is size XXLARGE</Text>
      <br />
      <Text size={TextSize.XLARGE}>This is size XLARGE</Text>
      <br />
      <Text size={TextSize.LARGE}>This is size LARGE</Text>
      <br />
      <Text size={TextSize.MEDIUM}>This is size MEDIUM</Text>
      <br />
      <Text size={TextSize.MEDIUM_SUB}>This is size MEDIUM_SUB</Text>
      <br />
      <Text size={TextSize.SMALL}>This is size SMALL</Text>
      <br />
      <Text size={TextSize.XSMALL}>This is size XSMALL</Text>
      <br />
    </div>
  ))
  .add('sizes (long text)', () => (
    <div>
      <Text>
        This is text without size specified. It will inherit its size and line-height from its
        parent. This is text without size specified. <strong>It will inherit</strong> its size and
        line-height from its parent. This is text without size specified. It will inherit its size
        and line-height from its parent.
      </Text>
      <br />
      <br />
      <Text size={TextSize.XXLARGE}>
        This is size XXLARGE. This is size XXLARGE. This is size XXLARGE. This is size XXLARGE. This
        is size XXLARGE. This is size XXLARGE. <strong>This is size XXLARGE.</strong> This is size
        XXLARGE. This is size XXLARGE. This is size XXLARGE. This is size XXLARGE. This is size
        XXLARGE.
      </Text>
      <br />
      <br />
      <Text size={TextSize.XLARGE}>
        This is size XLARGE. This is size XLARGE. This is size XLARGE. This is size XLARGE. This is
        size XLARGE. This is size XLARGE. <strong>This is size XLARGE.</strong> This is size XLARGE.
        This is size XLARGE. This is size XLARGE. This is size XLARGE. This is size XLARGE.
      </Text>
      <br />
      <br />
      <Text size={TextSize.LARGE}>
        This is size LARGE. This is size LARGE. This is size LARGE. This is size LARGE. This is size
        LARGE. This is size LARGE. <strong>This is size LARGE.</strong> This is size LARGE. This is
        size LARGE. This is size LARGE. This is size LARGE. This is size LARGE.
      </Text>
      <br />
      <br />
      <Text size={TextSize.MEDIUM}>
        This is size MEDIUM. This is size MEDIUM. This is size MEDIUM. This is size MEDIUM. This is
        size MEDIUM. This is size MEDIUM. <strong>This is size MEDIUM.</strong> This is size MEDIUM.
        This is size MEDIUM. This is size MEDIUM. This is size MEDIUM. This is size MEDIUM.
      </Text>
      <br />
      <br />
      <Text size={TextSize.MEDIUM_SUB}>
        This is size MEDIUM_SUB. This is size MEDIUM_SUB. This is size MEDIUM_SUB. This is size
        MEDIUM_SUB. This is size MEDIUM_SUB. This is size MEDIUM_SUB.{' '}
        <strong>This is size MEDIUM_SUB.</strong> This is size MEDIUM_SUB. This is size MEDIUM_SUB.
        This is size MEDIUM_SUB. This is size MEDIUM_SUB. This is size MEDIUM_SUB.
      </Text>
      <br />
      <br />
      <Text size={TextSize.SMALL}>
        This is size SMALL. This is size SMALL. This is size SMALL. This is size SMALL. This is size
        SMALL. This is size SMALL. <strong>This is size SMALL.</strong> This is size SMALL. This is
        size SMALL. This is size SMALL. This is size SMALL. This is size SMALL.
      </Text>
      <br />
      <br />
      <Text size={TextSize.XSMALL}>
        This is size XSMALL. This is size XSMALL. This is size XSMALL. This is size XSMALL. This is
        size XSMALL. This is size XSMALL. <strong>This is size XSMALL.</strong> This is size XSMALL.
        This is size XSMALL. This is size XSMALL. This is size XSMALL. This is size XSMALL.
      </Text>
    </div>
  ))
  .add('colors', () => (
    <div>
      <Text>
        This is text without a color specified. It will inherit its color from its parent.
      </Text>
      <br />
      <Text color={TextColor.PRIMARY}>This is color PRIMARY</Text>
      <br />
      <Text color={TextColor.SECONDARY}>This is color SECONDARY</Text>
      <br />
      <Text color={TextColor.METADATA}>This is color METADATA</Text>
      <br />
      <div style={{ backgroundColor: '#a8b0bd', padding: '16px' }}>
        <Text color={TextColor.WHITE}>
          This is color WHITE. It should be used where the background is a darker color.
        </Text>
      </div>
    </div>
  ));
