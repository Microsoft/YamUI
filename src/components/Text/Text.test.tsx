/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Text, { TextProps, TextColor, TextSize } from '.';

describe('<Text />', () => {
  let component: ShallowWrapper<TextProps, {}>;

  describe('with default options', () => {
    beforeEach(() => {
      component = shallow(<Text>test content</Text>);
    });

    it('renders its given content', () => {
      expect(component.text()).toEqual('test content');
    });

    it('contains its base className', () => {
      expect(component.hasClass('y-text')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<Text className="TEST_CLASSNAME">test content</Text>);
    });

    it('includes that className', () => {
      expect(component.hasClass('TEST_CLASSNAME')).toBe(true);
    });

    it('still includes its base className', () => {
      expect(component.hasClass('y-text')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with a valid size', () => {
    beforeEach(() => {
      component = shallow(<Text size={TextSize.XLARGE}>test content</Text>);
    });

    it('renders the correct size className', () => {
      expect(component.hasClass('y-textSize-xLarge')).toBe(true);
    });

    it('still has its base className', () => {
      expect(component.hasClass('y-text')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with a valid color', () => {
    beforeEach(() => {
      component = shallow(<Text color={TextColor.SECONDARY}>test content</Text>);
    });

    it('renders the correct color className', () => {
      expect(component.hasClass('y-text__color-secondary')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
