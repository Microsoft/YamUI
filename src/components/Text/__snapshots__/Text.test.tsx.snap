// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> with a valid color matches its snapshot 1`] = `
<span
  className="y-text y-text__color-secondary"
>
  test content
</span>
`;

exports[`<Text /> with a valid size matches its snapshot 1`] = `
<span
  className="y-text y-textSize-xLarge"
>
  test content
</span>
`;

exports[`<Text /> with additional className matches its snapshot 1`] = `
<span
  className="y-text TEST_CLASSNAME"
>
  test content
</span>
`;

exports[`<Text /> with default options matches its snapshot 1`] = `
<span
  className="y-text"
>
  test content
</span>
`;
